code/modules/detectivework/scanner.dm

```diff
@@ -30,16 +60,75 @@
 		scanner.displayDetectiveScanResults(usr)
 
 /obj/item/detective_scanner/attack_self(mob/user)
+	if(scanning)
+		to_chat(user, "<span class='notice'>[src] is in use.</span>")
+		return
+	radial_generate()
+	if(icons_available)
+		var/selection = show_radial_menu(user, src, icons_available, radius = 38, require_near = TRUE, tooltips = TRUE)
+		if(!selection)
+			return
+		
+		if(selection == "View results")
+			var/obj/item/detective_scanner/scanner = src
+			if(istype(scanner))
+				scanner.displayDetectiveScanResults(user)
```

Comment:
What else is src going to be in /obj/item/detective_scanner/attack_self() ?
```suggestion
			displayDetectiveScanResults(user)
```

Reply:
This was the previous code except copy-pasted.

Reply:
Point still stands